### PR TITLE
Finish off new 'packages' concept (#1586)

### LINKED ISSUE #1586: party-robot: teach package-import concept
The `party-robot` expects students to use the `fmt` package, but its [introduction.md document](https://github.com/exercism/go/blob/main/exercises/concept/party-robot/.docs/introduction.md) nor its dependencies explain how to import a package using the `import` keyword. 

This exercise should explain how to import packages in the `party-robot` exercise. This means updating the `introduction.md` file, adding a concept to the `concepts` array and creating the files for that concept.

### PR DESCRIPTION
**Pull request by bobtfish**
Fixes #1586

### CODE REVIEW

File: concepts/packages/.meta/config.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "blurb": "TODO: add blurb",
   "authors": ["ErikSchierboom"],
-  "contributors": []
+  "contributors": ["bobtfish"]
```
Comment (jmrunkle) [inline concepts/packages/.meta/config.json:4]: Personal opinion, your username belongs in the authors line since you added so much material here for the concept. In contrast, for the exercise this did not change much, but for the concept documentation it added over half of the material.

File: concepts/packages/.meta/config.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "blurb": "TODO: add blurb",
```
Comment (jmrunkle) [inline concepts/packages/.meta/config.json:2]: Can we complete this TODO?

File: concepts/packages/about.md [hunk 3]
```diff
@@ -6,7 +6,29 @@ In Go an application is organized in packages. A package is a collection of sour
...
+```
+
+Multiple packages can be imported at once by encoding them in brackets.
+
+```go
+package greeting
+
+import (
+	"errors"
+    "fmt"
```
Comment (jmrunkle) [inline concepts/packages/about.md:25]: indentation here?

File: concepts/packages/introduction.md [hunk 4]
```diff
@@ -1,52 +1,27 @@
...
-TODO: the content below is copied from the exercise introduction and probably needs rewriting to a proper concept introduction
-
-## packages
-
-## functions
-
-## variables
-
-In Go an application is organized in packages. A package is a collection of source files located in the same folder. All source files in a folder must have the same package name at the top of the file. The package name is preferred to be the same as the folder it is located in.
+In Go an application is organized in packages. A package is a collection of source files located in the same folder. All source files in a folder must have the same package name at the top of the file. The package is preferred to be the same as the folder it is located in.
```
Comment (jmrunkle) [inline concepts/packages/introduction.md:3]: Can we break this up into one line per sentence so that the blast radius for later changes is reduced?

```
In Go an application...
A package is...
All source files...
The package is preferred...
```

I think this was copy-pasted so could you fix the source as well?

File: concepts/packages/links.json [hunk 5]
```diff
@@ -1,35 +1,10 @@
 [
   {
-    "url": "https://golang.org/ref/spec#Assignments",
-    "description": "assignment"
+    "url": "https://golang.org/ref/spec#Import_declarations",
+    "description": "import-declaration"
```
Comment (jmrunkle) [inline concepts/packages/links.json:4]: The descriptions can be a little more meaningful like "import-declaration-spec"

Is the spec really the best links we can give them? I suspect there are good links to "A Tour of Go" which would be better: https://tour.golang.org/

File: concepts/packages/links.json [hunk 6]
```diff
@@ -1,35 +1,10 @@
...
-    "url": "https://golang.org/ref/spec#Return_statements",
-    "description": "return"
-  },
-  {
-    "url": "https://golang.org/ref/spec#Exported_identifiers",
-    "description": "public-vs-private"
-  },
-  { "url": "https://golang.org/ref/spec#Comments", "description": "comments" }
+    "url": "https://golang.org/ref/spec#Package_clause",
+    "description": "package"
```
Comment (jmrunkle) [inline concepts/packages/links.json:8]: package-spec

**Review by jmrunkle — CHANGES_REQUESTED** [review]
Thanks for working on this!

File: concepts/variables/introduction.md [hunk 7]
```diff
@@ -1,6 +1,9 @@
 # Introduction
 
-In Go an application is organized in packages. A package is a collection of source files located in the same folder. All source files in a folder must have the same package name at the top of the file. The package name is preferred to be the same as the folder it is located in.
+In Go an application is organized in packages.
+A package is a collection of source files located in the same folder.
+All source files in a folder must have the same package name at the top of the file.
+The package name is preferred to be the same as the folder it is located in.
```
Comment (junedev) [inline concepts/variables/introduction.md:6]: Why does the concept `variables` and `functions` both contain this section on packages? Shouldn't that be removed now that there is a package concept?

File: concepts/packages/.meta/config.json [hunk 8]
```diff
@@ -1,5 +1,5 @@
 {
-  "blurb": "TODO: add blurb",
-  "authors": ["ErikSchierboom"],
+  "blurb": "Packages are the unit of code reuse in Go",
+  "authors": ["ErikSchierboom", "bobtfish"],
```
Comment (ErikSchierboom) [inline concepts/packages/.meta/config.json:3]: I think I should probably be removed as an author:

```suggestion
  "authors": ["bobtfish"],
```

File: concepts/functions/introduction.md [hunk 9]
```diff
@@ -8,7 +8,10 @@ TODO: the content below is copied from the exercise introduction and probably ne
 
 ## variables
 
-In Go an application is organized in packages. A package is a collection of source files located in the same folder. All source files in a folder must have the same package name at the top of the file. The package name is preferred to be the same as the folder it is located in.
+In Go an application is organized in packages.
+A package is a collection of source files located in the same folder.
+All source files in a folder must have the same package name at the top of the file.
+The package name is preferred to be the same as the folder it is located in.
```
Comment (ErikSchierboom) [inline concepts/functions/introduction.md:14]: I'd remove this section as it seems to be unrelated to variables.

```suggestion
In Go an application is organized in packages.
A package is a collection of source files located in the same folder.
All source files in a folder must have the same package name at the top of the file.
The package name is preferred to be the same as the folder it is located in.
```

File: concepts/variables/introduction.md (hunk 7 above)
Comment (ErikSchierboom) [reply concepts/variables/introduction.md:6]: This section can indeed be removed from this document.

File: concepts/packages/introduction.md [hunk 10]
```diff
@@ -1,52 +1,30 @@
...
-count = 2  // Update to new value
+package greeting
 
-// Compiler error when assigning different type
-// count = false
+import "fmt"
 ```
 
-A function can have zero or more parameters. All parameters must be explicitly typed, there is no type inference for parameters. A function can also have multiple return values which must also be explicitly typed. Values are returned from functions using the `return` keyword. To allow a function to be called by code in other packages, the name of the function must start with a capital letter.
+Multiple packages can be imported at once by encoding them in brackets.
```
Comment (ErikSchierboom) [inline concepts/packages/introduction.md:21]: The importing of multiple packages is not needed to solve the exemplar solution. Quoting [the docs](https://exercism.org/docs/building/tracks/concept-exercises):

> Only information that is needed to understand the fundamentals of the concept and solve the exercise should be provided. Extra information should be left for the concept's about.md document.

This content is perfect for the `about.md` file, but should be removed from this `introduction.md` file.

**Review by ErikSchierboom — COMMENTED** [review]
Great! I've left some minor comments.

File: concepts/packages/introduction.md [hunk 11]
```diff
@@ -1,52 +1,20 @@
...
-count := 1 // Assign initial value
-count = 2  // Update to new value
-
-// Compiler error when assigning different type
-// count = false
-```
-
-A function can have zero or more parameters. All parameters must be explicitly typed, there is no type inference for parameters. A function can also have multiple return values which must also be explicitly typed. Values are returned from functions using the `return` keyword. To allow a function to be called by code in other packages, the name of the function must start with a capital letter.
+Go provides an extensive standard library of packages which you can use in your program using the `import` keyword.
+Standard library packages are imported using their name.
```
Comment (ErikSchierboom) [inline concepts/packages/introduction.md:13]: Do we also need to describe how things from imported packages can be used? Like `fmt.Sprintf` (with the `fmt.` prefix)?

Comment (bobtfish) [reply concepts/packages/introduction.md:13]: good call. I'll add a few more sentences later :)

File: concepts/packages/.meta/config.json [hunk 12]
```diff
@@ -1,5 +1,5 @@
 {
-  "blurb": "TODO: add blurb",
-  "authors": ["ErikSchierboom"],
+  "blurb": "Packages are the unit of code reuse in Go",
```
Comment (jmrunkle) [inline concepts/packages/.meta/config.json:2]:
```suggestion
  "blurb": "Packages are the unit of code reuse in Go.",
```

minor nit: I believe we have been using complete sentences ending in a period.

File: concepts/packages/about.md [hunk 13]
```diff
@@ -1,12 +1,37 @@
...
+Go provides an extensive standard library of packages which you can use in your program using the `import` keyword.
+Standard library packages are imported using their name.
+
+```go
+package greeting
+
+import "fmt"
+```
+
+Multiple packages can be imported at once by encoding them in brackets.
```
Comment (jmrunkle) [inline concepts/packages/about.md:21]:
```suggestion
Multiple packages can be imported at once by encoding them in parentheses.
```

I believe `[` is the (square) bracket, `(` is a parenthesis, `{` is a brace or curly bracket, and `<` is an inequality sign or pointy bracket. These are parentheses.

**Review by jmrunkle — APPROVED** [review]
Small nits, otherwise LGTM
